### DISCUSSION TITLE
How do I use this in a custom test type?

### DISCUSSION THREAD

**Discussion by jamesforwardnwboxed** [micronaut-projects/micronaut-test-resources#752] [Q&A]
Using Gradle, I've managed to get this working in the usual `test` execution, however I like having separate `test` (for pure unit tests) and `integrationTest` tasks. I know this shouldn't be strictly necessary in Micronaut, but this is just a personal preference to have them separated. It seems like micronaut test resources only spins up in the default `test` task execution. How do I make it work with my integrationTest task?

I've tried adding `.dependsOn internalStartTestResourcesService` which does seem to start up the internal test resource service, but does not seem to actually start containers or do any kind of property injection.

**Comment by sdelamo** [comment]
ping @melix

**Comment by melix** [comment]
The plugin [configures test resources automatically](https://github.com/micronaut-projects/micronaut-gradle-plugin/blob/7b4c61f1cc5c23d39ac87b0c11571527bf9a5f41/test-resources-plugin/src/main/java/io/micronaut/gradle/testresources/MicronautTestResourcesPlugin.java#L163) for all `Test` tasks, there's nothing particular you should have to deal with.
